web/js/modules/layers/selectors.js

```diff
@@ -590,6 +590,44 @@ export const subdailyLayersActive = createSelector(
   (layers) => hasSubDaily(layers),
 );
 
+/**
+ * Get subdaily layers from given layers
+ * @param {Array} layers
+ */
+export function getSubDaily(layers) {
+  const outputLayers = [];
+  if (layers && layers.length) {
+    for (let i = 0; i < layers.length; i += 1) {
+      if (layers[i].period === 'subdaily') {
+        outputLayers.push(layers[i]);
+      }
+    }
+  }
+  return outputLayers;
+}
+
+export const subdailyLayers = createSelector(
+  [getActiveLayers],
+  (layers) => getSubDaily(layers),
+);
+
+/**
+ * Gets smallest interval value of subdaily layers
+ * @param {Object} state
+ */
+export function getSmallestIntervalValue(state) {
+  const layers = getActiveLayers(state);
+  let smallestDelta = 1440;
+  if (layers && layers.length) {
```

Comment:
I would refactor this to only call `lodashGet(layers[i], 'dateRanges[0].dateInterval')` once.

 ```if (layers && layers.length) {
  for (let i = 0; i < layers.length; i += 1) {
    let interval = lodashGet(layers[i], 'dateRanges[0].dateInterval');
    if (layers[i].period === 'subdaily' && interval < smallestDelta) {
      smallestDelta = Number(interval);
    }
  }
}